src/test/java/com/onfido/integration/WorkflowRunTest.java

```diff
@@ -1,43 +1,102 @@
 import java.nio.file.Files;
+import java.util.HashMap;
+import java.util.Map;
 import java.util.UUID;
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 public class WorkflowRunTest extends TestBase {
+  private UUID applicantId;
   private WorkflowRun workflowRun;
-  private Applicant applicant;
+  private UUID workflowRunId;
 
   static final UUID WORKFLOW_ID = UUID.fromString("e8c921eb-0495-44fe-b655-bcdcaffdafe5");
 
   @BeforeEach
   public void setup() throws Exception {
-    applicant = createApplicant();
-    workflowRun = createWorkflowRun(WORKFLOW_ID, applicant.getId());
+    applicantId = createApplicant().getId();
+    workflowRun = createWorkflowRun(WORKFLOW_ID, applicantId);
+    workflowRunId = workflowRun.getId();
   }
 
   @Test
   public void createWorkflowRunTest() throws Exception {
     Assertions.assertEquals(WORKFLOW_ID, workflowRun.getWorkflowId());
-    Assertions.assertEquals(applicant.getId(), workflowRun.getApplicantId());
+    Assertions.assertEquals(applicantId, workflowRun.getApplicantId());
+  }
+
+  @Test
+  public void createWorkflowRunWithCustomInputsTest() throws Exception {
+    UUID workflowId = UUID.fromString("45092b29-f220-479e-aa6f-a6f989baac4c");
+    Map<String, Object> customData = new HashMap<>();
+    customData.put("age", 18);
+    customData.put("is_employed", true);
+
+    WorkflowRun workflowRunWithCustomInputs =
+        onfido.createWorkflowRun(
+            new WorkflowRunBuilder()
+                .workflowId(workflowId)
+                .applicantId(applicantId)
+                .customData(customData));
+
+    Assertions.assertEquals(workflowId, workflowRunWithCustomInputs.getWorkflowId());
+    Assertions.assertEquals(applicantId, workflowRunWithCustomInputs.getApplicantId());
+    Assertions.assertEquals(
+        WorkflowRun.StatusEnum.APPROVED, workflowRunWithCustomInputs.getStatus());
   }
 
   @Test
   public void findWorkflowRunTest() throws Exception {
-    WorkflowRun lookupWorkflowRun = onfido.findWorkflowRun(workflowRun.getId());
+    WorkflowRun lookupWorkflowRun = onfido.findWorkflowRun(workflowRunId);
 
     Assertions.assertEquals(WORKFLOW_ID, lookupWorkflowRun.getWorkflowId());
-    Assertions.assertEquals(applicant.getId(), lookupWorkflowRun.getApplicantId());
+    Assertions.assertEquals(applicantId, lookupWorkflowRun.getApplicantId());
   }
 
   @Test
   public void evidenceWorkflowRunTest() throws Exception {
-    File download = onfido.downloadSignedEvidenceFile(workflowRun.getId());
+    File download = onfido.downloadSignedEvidenceFile(workflowRunId);
+    byte[] content = Files.readAllBytes(download.toPath());
+
+    Assertions.assertEquals("%PDF", new String(content, 0, 4));
+    Assertions.assertTrue(download.length() > 0);
+  }
+
+  @Test
+  public void generateTimelineFileTest() throws Exception {
+    UUID workflowId = UUID.fromString("221f9d24-cf72-4762-ac4a-01bf3ccc09dd");
```

Comment:
What about storing this UUID (repeated a few lines below) in a constant?

Reply:
Good point 👍

Reply:
As discussed, it's fine to come back on this point in a subsequent PR.